packages/cli/src/public-api/v1/handlers/users/users.handler.ee.ts

```diff
@@ -98,7 +99,19 @@ export = {
 		isLicensed('feat:advancedPermissions'),
 		globalScope('user:changeRole'),
 		async (req: ChangeRole, res: Response) => {
-			await Container.get(UsersController).changeGlobalRole(req);
+			const validation = RoleChangeRequestDto.safeParse(req.body);
+			if (validation.error) {
+				return res.status(400).json({
+					message: validation.error.errors[0],
+				});
+			}
```

Comment:
We have to be careful as we add this to the Public API, e.g. if the validation is ever stricter than it used to be, it'll be a breaking change.

Reply:
true, but this specific validation used to be inside the controller using the decorated validation classes. The updated validation in this case is identical as before.